ext/testlib/terminal.py

```diff
@@ -137,35 +138,35 @@ def insert_separator(inside, char=default_separator,
     .. seealso:: :func:`separator`
     '''
     # Use a bytearray so it's efficient to manipulate
-    string = bytearray(separator(char, color=color))
+    string = bytearray(separator(char, color=color), 'utf-8')
 
     # Check if we can fit inside with at least min_barrier.
     gap = (len(string) - len(inside)) - min_barrier * 2
     if gap > 0:
         # We'll need to expand the string to fit us.
         string.extend([ char for _ in range(-gap)])
     # Emplace inside
-    middle = ((len(string)-1)/2)
-    start_idx = middle - len(inside)/2
-    string[start_idx:len(inside)+start_idx] = inside
-    return str(string)
+    middle = int((len(string)-1)/2)
+    start_idx = int(middle - len(inside)/2)
```

Comment:
Also, your change isn't the same. The parentheses are in the wrong places.

```suggestion
    start_idx = middle - len(inside)//2
```